components/script/dom/abortsignal.rs

```diff
@@ -139,7 +139,17 @@ impl AbortSignalMethods<crate::DomTypeHolder> for AbortSignal {
     /// <https://dom.spec.whatwg.org/#dom-abortsignal-throwifaborted>
     #[allow(unsafe_code)]
     fn ThrowIfAborted(&self) {
-        // TODO
+        // The throwIfAborted() method steps are to throw this’s abort reason, if this is aborted.
+        if self.aborted() {
+            let cx = GlobalScope::get_cx();
+            unsafe {
+                JS_SetPendingException(
+                    *cx,
+                    self.abort_reason.handle(),
+                    ExceptionStackBehavior::Capture,
+                )
+            };
+        }
```

Comment:
I would have expected that the method would need to return a result, with `Error:JSFailed` used when the pending exception is set.

Reply:
I think we should just note it in a issue for now, and check it later as it comes up when we integrate.